### PR TITLE
refactored users api

### PR DESCRIPTION
**Pull request by Dhroov7**
@championswimmer Please review.

### CODE REVIEW

**Comment by championswimmer** [comment]
This pull request **introduces 2 alerts** when merging 0ca338142004b8b0a5f78df6d1c4c36e60b42bb4 into 2dd2301a603fdb01cfcb44c54643d350eb58aa73 - [view on LGTM.com](https://lgtm.com/projects/g/coding-blocks/oneauth/rev/pr-5623a4e05319e314e5ca53059b4645b919219fca)

**new alerts:**

* 2 for Unused variable, import, function or class

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by championswimmer** [comment]
This pull request **introduces 5 alerts** when merging 904ab8c26ce35bccd2977012d49234760c6d52d0 into 2dd2301a603fdb01cfcb44c54643d350eb58aa73 - [view on LGTM.com](https://lgtm.com/projects/g/coding-blocks/oneauth/rev/pr-1759e9d7aaa307c1d4d797b8324041c94d73c8fd)

**new alerts:**

* 5 for Unused variable, import, function or class

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: src/middlewares/acl.js [hunk 1]
```diff
@@ -16,7 +16,16 @@ function ensureRole(role) {
     }
 }
 
+function ensureTrustedClient(req, res, next) {
+    if(req.client && req.client.trusted) {
+        next()
+    }else{
+        res.status(403).render('error')
```
Comment (championswimmer) [inline src/middlewares/acl.js]: error should be JSON object 

```
{ status: 4000,
  message: 'Only for trusted client'
}
```

File: src/routers/api/users/index.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    let password = await passutils.pass2hash(uid(30))
+
+    const userLocal = await createUserLocal(query, password, includes)
+
+    let key = uid(15)
+    mail.forgotPassEmail(userLocal, key)
+
+    res.send(userLocal)
+  } catch (err) {
+    res.send('Error while creating user.')
```
Comment (championswimmer) [inline src/routers/api/users/index.js]: with 400 status code, proper error as object

File: src/routers/api/users/id.js [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+    }
+)
+
+router.post('/:id/address', ensureTrustedClient, async (req, res) => {
+
+    if (hasNull(req.body, ['first_name', 'last_name', 'number', 'email', 'pincode', 'street_address', 'landmark', 'city', 'stateId', 'countryId', 'dial_code'])) {
+        return res.send(400)
+    } else {
+        if (!req.body.label) {
+            return red.status(400)
```
Comment (championswimmer) [inline src/routers/api/users/id.js]: res not red

File: src/routers/api/users/id.js [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+        }
+
+        if (!req.body.dial_code) {
+            return res.status(400)
```
Comment (championswimmer) [inline src/routers/api/users/id.js]: errors are not being sent. 

also send a JSON object response

File: src/routers/api/users/id.js [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+        }
+
+        try {
+            const user = await findUserById(req.params.id, null)
+            if (!user) {
+                return res.send('User not found.')
```
Comment (championswimmer) [inline src/routers/api/users/id.js]: 404 status

Comment (championswimmer) [reply src/routers/api/users/id.js]:
```js
res.status(404).send({
   messsage: 'User not found'
})

File: src/routers/api/users/id.js [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+                dial_code: req.body.dial_code,
+                demographicId: demographics.id,
+                whatsapp_number: req.body.whatsapp_number || null,
+                // if no addresses, then first one added is primary
+                primary: !demographics.addresses
+            }
+
+            let number = options.dial_code + options.mobile_number
+            if (!validateNumber(parseNumberByCountry(number, req.body.countryId))) {
+                return res.send(400)
```
Comment (championswimmer) [inline src/routers/api/users/id.js]:
```js
res.status(400).send({message: 'invalid mobile' }
```

File: src/routers/api/users/index.js [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+    }
+  }
+)
+
+router.post('/', ensureTrustedClient, async (req, res) => {
+
+  let user = {}
+
+  if (!req.body.username) {
+    res.status(400)
```
Comment (championswimmer) [inline src/routers/api/users/index.js]: should be send too, not just status

File: src/routers/api/users/index.js [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+  if (!req.body.mobile_number) {
+    res.status(400)
+  }
+
+  if (!req.body.dial_code) {
+    res.status(400)
+  }
+
+  try {
+    let user_exist = await findUserByParams({ username: req.body.username })
```
Comment (championswimmer) [inline src/routers/api/users/index.js]: use camelCase (only DB columns are underscore case)

File: src/middlewares/acl.js [hunk 9]
```diff
@@ -20,7 +20,7 @@ function ensureTrustedClient(req, res, next) {
     if(req.client && req.client.trusted) {
         next()
     }else{
-        res.status(403).render('error')
+        res.status(400).send({error: 'Only for trusted client.'})
```
Comment (championswimmer) [inline src/middlewares/acl.js]: code was 403 correct earlier

**Comment by championswimmer** [comment]
This pull request **introduces 5 alerts** when merging 526684e85194da8bb3b109d531b792374a41464a into 68e45a1c054d63f9359bea618ccfc06b785afe9a - [view on LGTM.com](https://lgtm.com/projects/g/coding-blocks/oneauth/rev/pr-052283943201878a18ee7f962a6a577dbcae83d0)

**new alerts:**

* 5 for Unused variable, import, function or class

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by Dhroov7** [comment]
@championswimmer any updates on this PR?

**Pull request closed without merging**